api/src/opentrons/protocol_api/labware.py

```diff
@@ -341,16 +342,18 @@ def uri(self) -> str:
 
     @property  # type: ignore[misc]
     @requires_version(2, 0)
-    def parent(self) -> Union[str, ModuleTypes, None]:
+    def parent(self) -> Union[str, ModuleTypes, OffDeckType]:
         """The parent of this labware.
 
         If the labware is on the deck, a `str` deck slot name will be returned.
         If on a module, the parent :py:class:`ModuleContext` will be returned.
-        If off deck, `None` will be returned.
+        If off deck, `off-deck` will be returned.
 
         .. versionchanged:: 2.14
             Return type for module parent changed to :py:class:`ModuleContext`.
             Prior to this version, an internal geometry interface is returned.
+        .. versionchanged:: 2.15
+            Return type for off-deck changed from None to off-deck.
```

Comment:
As far as I can tell, there has never been a public release of the Python Protocol API where people would see `None` returned for off-deck labware.

But I just ran this quick experiment, which shows that they would have seen where the labware used to be:

```python
metadata = {
    "apiLevel": "2.13",
}

def run(protocol) -> None:
    l = protocol.load_labware("opentrons_96_tiprack_300ul", 1)
    print("Before del:", l.parent)  # prints "1"
    del protocol.deck[1]
    print("After del:", l.parent)  # prints "1" again
```

Weird! We may as well document it.

```suggestion
        .. versionchanged:: 2.15
            Will now return :py:obj:`OFF_DECK` if the labware is off-deck.
            Formerly, if the labware was removed by using ``del`` on :py:obj:`.deck`,
            this would return where it was before its removal.
```